api/api/src/main/scala/uk/ac/wellcome/platform/api/Router.scala

```diff
@@ -221,6 +221,14 @@ class Router(elasticClient: ElasticClient,
       }
     }
 
+  def extractPublicUri =
+    extractUri.map { uri =>
+      uri
+        .withHost(apiConfig.host)
+        .withPort(0) // Port 0 is used here to indicate to akka no explicit port
+        .withScheme(apiConfig.scheme)
+    }
```

Comment:
Perhaps worth including the comment from the PR description explaining why we're doing this, and what the consequences of not doing it are? With examples of before/after URIs?

Reply:
:+1: done

Reply:
What should the URIs look like after you've done this? Also I liked the comment about `withPort(0)` being a way to tell Akka not to set an explicit port.

Reply:
Have updated, what about now?